benchmarks/TFLite/android-adreno.cmake

```diff
@@ -0,0 +1,138 @@
+
+set(ANDROID_ADRENO_GPU_TRANSLATION_FLAGS
+  "--iree-input-type=tosa"
+  "--iree-vulkan-target-triple=adreno-unknown-android11"
+)
+
+# GPU, Vulkan, Adreno, full-inference
+iree_benchmark_suite(
+  SUITE_NAME
+    "android-adreno"
+
+  MODULES
+    "${DEEPLABV3_FP32_MODULE}"
+    "${MOBILESSD_FP32_MODULE}"
+    "${POSENET_FP32_MODULE}"
+    "${MOBILEBERT_FP32_MODULE}"
+    "${MOBILENET_V2_MODULE}"
+    "${MOBILENET_V3SMALL_MODULE}"
+
+  BENCHMARK_MODES
+    "full-inference,default-flags"
+  TARGET_BACKEND
+    "vulkan-spirv"
+  TARGET_ARCHITECTURE
+    "GPU-Adreno"
+  TRANSLATION_FLAGS
+    ${ANDROID_ADRENO_GPU_TRANSLATION_FLAGS}
+  BENCHMARK_TOOL
+    iree-benchmark-module
+  DRIVER
+    "vulkan"
+)
+
+################################################################################
+
+################################################################################
+#                                                                              #
+# Specialized benchmark configurations                                         #
+#                                                                              #
+# Each suite benchmarks one or more module with configurations that can vary   #
+# on model or architecture characteristics. These are intended for providing   #
+# continuous benchmarks of experimental features that cannot be turned on by   #
+# default yet. It is primarily intended for whoever is actively investigating  #
+# optimizations for a feature exemplified in a specific model or architecture. #
+# Due to our current benchmark setup, there can only be one experimental       #
+# configuration per model and other benchmark mode.                            #
+#                                                                              #
+################################################################################
+
+# GPU, Vulkan, Adreno, full-inference
+iree_benchmark_suite(
+  SUITE_NAME
+    "android-adreno"
+
+  MODULES
+    "${DEEPLABV3_FP32_MODULE}"
+    "${MOBILESSD_FP32_MODULE}"
+    "${POSENET_FP32_MODULE}"
+    "${MOBILEBERT_FP32_MODULE}"
+    "${MOBILENET_V2_MODULE}"
+    "${MOBILENET_V3SMALL_MODULE}"
+
+  BENCHMARK_MODES
+    "full-inference,experimental-flags"
+  TARGET_BACKEND
+    "vulkan-spirv"
+  TARGET_ARCHITECTURE
+    "GPU-Adreno"
+  TRANSLATION_FLAGS
+    ${ANDROID_ADRENO_GPU_TRANSLATION_FLAGS}
+    "--iree-flow-enable-fuse-padding-into-consumer-ops"
+  BENCHMARK_TOOL
+    iree-benchmark-module
+  DRIVER
+    "vulkan"
+)
+
+# kernel-execution
+
+# Note that for kernel-execution benchmarks batch_size/repeat-count need to be
+# low enough that the whole dispatch completes within an OS-specific timeout.
+# Otherwise you'll get error like:
+# ```
+# INTERNAL; VK_ERROR_DEVICE_LOST; vkQueueSubmit; while invoking native function
+# hal.ex.submit_and_wait; while calling import;
+# ```
+# With current kernel performance and timeouts on Adreno Pixel 4, this means we
```

Comment:
Didn't we stop benchmarking on Pixel 4 and switch to another Adreno GPU? Not this PR, but we should see if these can be enabled. @antiagainst

Reply:
Ah, good catch. Yes, I just turned off Pixel 4 Vulkan benchmarking in https://github.com/google/iree/pull/8018. We still benchmark adreno + Vulkan on Moto Edge X30. We can revisit what this comment says, probably after this refactoring PR.

Reply:
Oh yeah, for sure. I just noticed it as a separate thing while reviewing this